### DISCUSSION TITLE
Set every Pixel's Color Component

### DISCUSSION THREAD

**Discussion by brianmed** [SixLabors/ImageSharp#2421] [Q&A] [answered]
I want to set all .A color component of a Rgba32, Bgra32, or Argb32 pixel format image to 255.  Is the ProcessPixelRows Api the best way to do this?  Considering I want all pixels to have the exact same value, would a custom IPixel class be better?

**Comment by antonfirsov** [comment, marked as answer]
Considering that your image needs to be fully opaque, can't you use `Rgb24` or `Bgr24` ? Otherwise `ProcessPixelRows` is the fastest, while [`ProcessPixelRowsAsVector4`](https://docs.sixlabors.com/articles/imagesharp/pixelbuffers.html#parallel-pixel-format-agnostic-image-manipulation) is a convenient pixel-agnostic way to do it.

**Reply by brianmed** [reply]
That's a good point about Rgb24 or Bgr24.  What I did not articulate is that there is a pipeline.  First, we set the .A color component to a known value and then we adjust as needed via Opacity.

**Reply by JimBobSquarePants** [reply]
>I want all pixels to have the exact same value, would a custom IPixel class be better

That wouldn't be useful in your scenario since you have a pipeline and the aforementioned pixel formats do the same thing.

We actually have a processor in the test project that does this operation. This should be fairly efficient (though could be SIMD optimized). I'd just copy that for now.

https://github.com/SixLabors/ImageSharp/blob/4989e146116e65c16ec26d63fe44ded6f1019c43/tests/ImageSharp.Tests/TestUtilities/TestImageExtensions.cs#L19-L24
